ntpd: fails to compile in master

Maintainer: @ffontaine @hauke @knthm 
Environment: any arch, master

Description:
Building a clean OpenWrt for e.g. x86 at commit decd5e0cff5efc666327714e5b83b11ceb52b93e and packages feed at f9b1adf19d4b71f799a474537a6adbf45b8263b6 fails with:

```
    ERROR: package/feeds/packages/ntpd failed to build.
```

The actual error is:

```
make world V=sc
[..]
checking for library containing inet_pton... none required
checking for library containing inet_ntop... none required
checking for library containing setsockopt... none required
checking for recvmsg... yes
checking for inline... inline
checking size of time_t... 0
checking whether char is unsigned... yes
checking size of signed char... 0
checking for s_char... no
checking for long long... no
checking size of short... (cached) 2
checking size of int... (cached) 4
checking size of long... (cached) 4
configure: error: No way to specify a signed character!
make[6]: *** [Makefile:717: config.status] Error 1
make[6]: Leaving directory '/home/bluecmd/openwrt-clean/build_dir/target-i386_pentium4_musl/ntp-4.2.8p17/sntp'
make[5]: *** [Makefile:665: all-recursive] Error 1
make[5]: Leaving directory '/home/bluecmd/openwrt-clean/build_dir/target-i386_pentium4_musl/ntp-4.2.8p17'
make[4]: *** [Makefile:597: all] Error 2
make[4]: Leaving directory '/home/bluecmd/openwrt-clean/build_dir/target-i386_pentium4_musl/ntp-4.2.8p17'
make[3]: *** [Makefile:162: /home/bluecmd/openwrt-clean/build_dir/target-i386_pentium4_musl/ntp-4.2.8p17/.built] Error 2
make[3]: Leaving directory '/home/bluecmd/openwrt-clean/feeds/packages/net/ntpd'
time: package/feeds/packages/ntpd/compile#2.53#1.58#3.97
    ERROR: package/feeds/packages/ntpd failed to build.
make[2]: *** [package/Makefile:177: package/feeds/packages/ntpd/compile] Error 1
make[2]: Leaving directory '/home/bluecmd/openwrt-clean'
make[1]: *** [package/Makefile:171: /home/bluecmd/openwrt-clean/staging_dir/target-i386_pentium4_musl/stamp/.package_compile] Error 2
make[1]: Leaving directory '/home/bluecmd/openwrt-clean'
make: *** [/home/bluecmd/openwrt-clean/include/toplevel.mk:248: world] Error 2
```

A somewhat minimal diffconfig to test with:

```
CONFIG_TARGET_x86=y
CONFIG_TARGET_x86_generic=y
CONFIG_TARGET_x86_generic_DEVICE_generic=y
CONFIG_OPENSSL_ENGINE=y
CONFIG_OPENSSL_OPTIMIZE_SPEED=y
CONFIG_OPENSSL_PREFER_CHACHA_OVER_GCM=y
CONFIG_OPENSSL_WITH_ASM=y
CONFIG_OPENSSL_WITH_CHACHA_POLY1305=y
CONFIG_OPENSSL_WITH_CMS=y
CONFIG_OPENSSL_WITH_DEPRECATED=y
CONFIG_OPENSSL_WITH_ERROR_MESSAGES=y
CONFIG_OPENSSL_WITH_IDEA=y
CONFIG_OPENSSL_WITH_MDC2=y
CONFIG_OPENSSL_WITH_PSK=y
CONFIG_OPENSSL_WITH_SEED=y
CONFIG_OPENSSL_WITH_SRP=y
CONFIG_OPENSSL_WITH_SSE2=y
CONFIG_OPENSSL_WITH_TLS13=y
CONFIG_OPENSSL_WITH_WHIRLPOOL=y
CONFIG_PACKAGE_libcap=y
CONFIG_PACKAGE_libevent2-core=y
CONFIG_PACKAGE_libevent2-pthreads=y
CONFIG_PACKAGE_libopenssl=y
CONFIG_PACKAGE_ntp-utils=y
CONFIG_PACKAGE_ntpd=y
```

This happens on Octeon and x86_64 as well which is how I detected it.

Full config.log attached: [config.log](https://github.com/user-attachments/files/16848832/config.log)

_Possibly_ cause by this error?

```

conftest.c:57:18: error: 'long long long' is too long for GCC
   57 | #define intmax_t long long int
      |                  ^~~~
conftest.c:57:23: error: 'long long long' is too long for GCC
   57 | #define intmax_t long long int
      |                       ^~~~
conftest.c:58:19: error: duplicate 'unsigned'
   58 | #define uintmax_t unsigned long long int
      |                   ^~~~~~~~
conftest.c:58:28: error: 'long long long' is too long for GCC
   58 | #define uintmax_t unsigned long long int
      |                            ^~~~
conftest.c:58:33: error: 'long long long' is too long for GCC
   58 | #define uintmax_t unsigned long long int
      |                                 ^~~~

```

--- a/net/ntpd/Makefile
+++ b/net/ntpd/Makefile
@@ -9,7 +9,7 @@ include $(TOPDIR)/rules.mk
 
 PKG_NAME:=ntp
 PKG_VERSION:=4.2.8p17
-PKG_RELEASE:=1
+PKG_RELEASE:=2
 
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
 PKG_SOURCE_URL:=http://www.eecis.udel.edu/~ntp/ntp_spool/ntp4/ntp-4.2/
@@ -19,7 +19,6 @@ PKG_LICENSE:=NTP
 PKG_LICENSE_FILES:=COPYRIGHT html/copyright.html
 PKG_CPE_ID:=cpe:/a:ntp:ntp
 
-PKG_FIXUP:=autoreconf
 PKG_LIBTOOL_PATHS:=. sntp
 PKG_CHECK_FORMAT_SECURITY:=0
 PKG_BUILD_PARALLEL:=1

